lib/screens/start_game/start_game_screen.dart

```diff
@@ -34,65 +34,67 @@ class _StartGameState extends State<StartGame> {
   void initState() {
     super.initState();
 
-    textFieldControllers.clear();
+    teamNameFieldsControllers.clear();
   }
 
   @override
   void dispose() {
     super.dispose();
 
-    textFieldControllers.forEach((controller) {
+    teamNameFieldsControllers.forEach((controller) {
       controller.dispose();
     });
   }
 
   void updateNumberOfTeams(int chosenValue) {
-    textFieldControllers.clear();
+    teamNameFieldsControllers.clear();
     setState(() {
       numOfTeamsValue = chosenValue;
     });
   }
 
-  void validateTextFields(textFieldControllers) {
-    var counter = 0;
+  void validateTeamNames(textFieldControllers) {
+    var errorsCount = 0;
     textFieldControllers.forEach((controller) {
-      if (controller.text.isEmpty) counter++;
+      if (controller.text.isEmpty) errorsCount++;
     });
     textFieldControllers.clear();
 
-    counter > 0 ? validated = false : validated = true;
+    errorsCount > 0 ? validated = false : validated = true;
   }
 
   NameOfTeam createNameOfTeamTextField(int index) {
-    textFieldControllers.add(TextEditingController());
+    // show previous team names, don't clear them on validation failure (e.g. missing team name)
+    String teamName = teamNames.length >= (index + 1) ? teamNames[index] : '';
+    teamNameFieldsControllers.add(TextEditingController(text: teamName));
```

Comment:
Added to prevent clearing team names in case when team validation failed (some team name is missing)